client/components/relatedProducts/Card.jsx

```diff
@@ -1,42 +1,57 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { AiOutlineStar } from 'react-icons/ai';
 
 import StarRating from '../StarRating';
 
+import getStarRatingAvg from '../../utils/getStarRatingAvg';
+import getImageUrl from '../../utils/getImageUrl';
+
 import '../../css/relatedProducts/Card.scss';
 
 const STAR_SIZE = 24;
 
 const Card = ({
-  relatedProduct, imageUrl, rating, handleCardClick,
-}) => (
-  <div className="related-products__card">
-    <div className="related-products__card-image" style={{ backgroundImage: `url(${imageUrl})` }}>
-      <span className="related-products__action-btn" role="button" tabIndex="0" onKeyPress={() => { handleCardClick(relatedProduct); }} onClick={() => handleCardClick(relatedProduct)}>
-        <AiOutlineStar size={STAR_SIZE} />
-      </span>
-    </div>
-    <div className="related-products__card-details">
-      <p className="related-products__card-category">{relatedProduct.category}</p>
-      <p className="related-products__card-expanded-name">
-        {`${relatedProduct.name} : ${relatedProduct.slogan}`}
-      </p>
-      <p className="related-products__card-price">{`$${relatedProduct.default_price}`}</p>
-      <StarRating size={STAR_SIZE} rating={rating} />
+  relatedProduct, handleActionBtnClick, handleCardClick,
+}) => {
+  const [rating, setRating] = useState(0);
+  const [imageUrl, setImageUrl] = useState('');
+  const hasNoRating = Number.isNaN(rating);
+
+  useEffect(() => {
+    getStarRatingAvg(relatedProduct.id).then((result) => setRating(result.avg));
+    getImageUrl(relatedProduct.id).then((url) => setImageUrl(url));
+  }, []);
+
+  return (
+    <div className="related-products__card" role="button" tabIndex={0} onKeyPress={() => { handleCardClick(relatedProduct.id); }} onClick={() => { handleCardClick(relatedProduct.id); }}>
+      <div className="related-products__card-image" style={imageUrl ? { backgroundImage: `url(${imageUrl})` } : {}}>
```

Comment:
This line uses a ternary to use the default background image in Card.scss when one isn't provided